earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -84,6 +169,8 @@ CreateEmptyDocument(const std::string &rootTagname) throw()
                                             0);// document type object (DTD)
     return doc;
    } catch (...) {
+     // if there is an error, unlock the document
+     if (doTerminate) xmlDocLock.Unlock();
```

Comment:
Just double checking - at one point we talked about just locking around the terminate and initialize functions. Did that not work?